core/domain/suggestion_services.py

```diff
@@ -0,0 +1,367 @@
+
+    Returns:
+        list(Suggestions): a list of suggestions reviewed by the given user.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_reviewed_by(
+            reviewer_id)]
+
+
+def get_suggestions_assigned_to_reviewer(assigned_reviewer_id):
+    """Gets a list of suggestions assigned to the given user for review.
+
+    Args:
+        assigned_reviewer_id: str. The reviewer assigned to review the
+                suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions assigned to the given user
+            for review.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel
+        .get_suggestions_assigned_to_reviewer(assigned_reviewer_id)]
+
+
+def get_suggestions_by_status(status):
+    """Gets a list of suggestions with the given status.
+
+    Args:
+        status: str. The status of the suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions with the given status.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_status(
+            status)]
+
+
+def get_suggestion_by_type(suggestion_type):
+    """Gets a list of suggestions with the given sub_type.
+
+    Args:
+        suggestion_type: str. The sub type of the suggestion.
+
+    Returns:
+        list(Suggestions): a list of suggestions of the given type.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_type(
+            suggestion_type)]
+
+
+def get_suggestions_by_target_id(target_type, target_id):
+    """Gets a list of suggestions to the entity with the given ID.
+
+    Args:
+        target_type: str. The type of target entity the suggestion is linked to.
+        target_id: str. The ID of the target entity the suggestion is linked to.
+
+    Returns:
+        list(Suggestions): a list of suggestions linked to the entity.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_target_id(
+            target_type, target_id)]
+
+
+def update_suggestion(suggestion):
+    """Updates the given sugesstion. The properties that can be edited are the
+    following:
+        status, reviewer_id, assigned_reviewer_id, payload.
+
+    Args:
+        suggesstion: Suggestion. The domain object of the suggestion to be
+            updated.
+    """
+
+    suggestion_model = suggestion_models.SuggestionModel.get_by_id(
+        suggestion.suggestion_id)
+
+    if suggestion_model.status == suggestion_models.STATUS_IN_REVIEW:
+        suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
+        suggestion_model.assigned_reviewer_id = suggestion.assigned_reviewer_id
+
+        suggestion_model.change_cmd = suggestion.change_cmd
```

Comment:
In which scenario would we able to update the `change_cmd`?

Reply:
The change_cmd has an 'old_value' field (for some types of suggestions) which we update to the content just before accepting the suggested value.

Also to allow changes to the suggestion after review, we need to allow the author to edit the suggestion (which is stored in the change_cmd). So then also the change_cmd will be updated.